dockertest/subtest.py

```diff
@@ -19,16 +19,58 @@
 from ConfigParser import Error
 from autotest.client.shared.error import TestError, TestNAError
 from autotest.client.shared.version import get_version
-from autotest.client import test
+from autotest.client import test, utils
 import version
 import config
 import subtestbase
+from docker_daemon import which_docker
 from xceptions import DockerTestFail
 from xceptions import DockerTestNAError
 from xceptions import DockerTestError
 from xceptions import DockerSubSubtestNAError
 
 
+MEMO_CACHE = {}
+
+
+def docker_rpm():
+    """
+    Returns the full NVRA of the currently-installed docker or docker-latest
+    """
+    if 'docker_rpm' not in MEMO_CACHE:
+        rpm = utils.run("rpm -q %s" % which_docker()).stdout.strip()
+        MEMO_CACHE['docker_rpm'] = rpm
+    return MEMO_CACHE['docker_rpm']
+
+
+def known_failures():
+    """
+    Returns a dict containing known test failures. Primary key is
+    subtest name (e.g. docker_cli/foo/bar), value is another dict
+    whose key is docker NVRA (e.g. docker-1.12.5-8.el7.x86_64),
+    value of that is a string description of the problem (e.g.
+    a bz number and comment).
+
+    FIXME: file path is hardcoded for development purposes; it
+    should be centrally accessible somewhere, possibly provided
+    at setup time by ADEPT.
+    """
+    if 'known_failures' not in MEMO_CACHE:
+        known = {}
+        import csv
+        # FIXME: how to determine path to top-level directory?
+        path_known = '/var/lib/autotest/client/tests/docker/known_failures.csv'
```

Comment:
How about ``config.CONFIGCUSTOMS``?  The config module ignores anything that doesn't end in ``.ini``.   It's handy to keep all the runtime configuration in one place for collection/archiving/reporting purposes.